### PR TITLE
Don't error when attempting to trigger schedules for inactive projects

### PR DESCRIPTION
**Pull request by katrogan**
## Why are the changes needed?
Currently the scheduler will fail its scheduling loop when any execution fails to trigger
see here: https://github.com/flyteorg/flyte/blob/a13a63dfd24470f566b11c7c107cc9655d3ec3c0/flyteadmin/scheduler/core/gocron_scheduler.go#L189 
which is called in the primary Run method here: https://github.com/flyteorg/flyte/blob/a13a63dfd24470f566b11c7c107cc9655d3ec3c0/flyteadmin/scheduler/schedule_executor.go#L82-L89
which results in a fatal log here: https://github.com/flyteorg/flyte/blob/a13a63dfd24470f566b11c7c107cc9655d3ec3c0/flyteadmin/scheduler/start.go#L58 which terminates the scheduler.

## What changes were proposed in this pull request?

This changes introduces a well structured error message type when a project already exists, and configures the scheduler to move on gracefully when an active scheduler is triggered for an archived project.

This mimics the existing behavior for the scheduler encountering an edge case when it attempts to trigger a formerly active schedule: https://github.com/flyteorg/flyte/blob/master/flyteadmin/scheduler/executor/executor_impl.go#L94

## How was this patch tested?

I ran flyte sandbox locally and activated a once minute launch plan. I archived the parent project (flytesnacks) and observed the scheduler continued to run.

```
{"json":{"src":"execution_manager.go:833"},"level":"debug","msg":"Failed to validate ExecutionCreateRequest {state:{NoUnkeyedLiterals:{} DoNotCompare:[] DoNotCopy:[] atomicMessageInfo:0x1400033c008} sizeCache:0 unknownFields:[] Project:flytesnacks Domain:development Name:f182916c04b5dc387000 Spec:launch_plan:{resource_type:LAUNCH_PLAN project:\"flytesnacks\" domain:\"development\" name:\"int_doubler_wf_scheduled\" version:\"kxtp6OI3ahFkpOj50oiQ2g\"} metadata:{mode:SCHEDULED scheduled_at:{seconds:1723467046}} Inputs:<nil> Org:} with err project [flytesnacks] is not active","ts":"2024-08-12T14:50:46+02:00"}
{"json":{"src":"auth_interceptor.go:155"},"level":"debug","msg":"Request failed due to [rpc error: code = InvalidArgument desc = project [flytesnacks] is not active]. If it's an unauthenticated error, we will attempt to establish an authenticated context.","ts":"2024-08-12T14:50:46+02:00"}
{"json":{"routine":"jobfunc-8988759870306525270","src":"executor_impl.go:118"},"level":"debug","msg":"project flytesnacks is inactive, ignoring schedule create failure for {BaseModel:{ID:2 CreatedAt:2024-08-12 14:47:44.00598 +0200 CEST UpdatedAt:2024-08-12 14:47:44.00598 +0200 CEST DeletedAt:<nil>} SchedulableEntityKey:{Project:flytesnacks Domain:development Name:int_doubler_wf_scheduled Version:kxtp6OI3ahFkpOj50oiQ2g} CronExpression: FixedRateValue:1 Unit:MINUTE KickoffTimeInputArg: Active:0x1400539c6d0}","ts":"2024-08-12T14:50:46+02:00"}
{"json":{"routine":"jobfunc-8988759870306525270","src":"executor_impl.go:129"},"level":"info","msg":"successfully fired the request for schedule {BaseModel:{ID:2 CreatedAt:2024-08-12 14:47:44.00598 +0200 CEST UpdatedAt:2024-08-12 14:47:44.00598 +0200 CEST DeletedAt:<nil>} SchedulableEntityKey:{Project:flytesnacks Domain:development Name:int_doubler_wf_scheduled Version:kxtp6OI3ahFkpOj50oiQ2g} CronExpression: FixedRateValue:1 Unit:MINUTE KickoffTimeInputArg: Active:0x1400539c6d0} for time 2024-08-12 14:50:46 +0200 CEST","ts":"2024-08-12T14:50:46+02:00"}


```

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

File: flyteidl/protos/flyteidl/admin/project.proto [hunk 1]
```diff
@@ -118,3 +118,13 @@ message ProjectGetRequest {
     // Optional, org key applied to the resource.
     string org = 2;
 }
+
+
+// Error returned for inactive projects
+message InactiveProject {
+    // Globally unique project name.
+    string id               = 1;
```
Comment (eapolinario) [inline flyteidl/protos/flyteidl/admin/project.proto:126]: nit: indent.

File: flyteidl/protos/flyteidl/admin/project.proto [hunk 2]
```diff
@@ -118,3 +118,13 @@ message ProjectGetRequest {
...
+
+    // Optional, org key applied to the resource.
+    string org = 5;
```
Comment (eapolinario) [inline flyteidl/protos/flyteidl/admin/project.proto:129]: why the gap in field numbers?

**Review by eapolinario — COMMENTED** [review]
Looks great, just a couple of nits.

File: flyteidl/protos/flyteidl/admin/project.proto (hunk 2 above)
Comment (katrogan) [reply flyteidl/protos/flyteidl/admin/project.proto:129]: copy paste 😅

**Comment by katrogan** [comment]
PTAL @eapolinario

**Comment by katrogan** [comment]
PTAL @eapolinario

**Review by eapolinario — APPROVED** [review]

**Pull request merged**